.github/workflows/integration-pr.yml

```diff
@@ -0,0 +1,196 @@
+# **This is not a trusted workflow.**
+# **what?** - This workflow runs integration tests for supported OS,
+# python versions and core adapters. It will only run tests for
+# related file changes. Use the `test all` and `test ${adapter}`
```

Comment:
Does "related file changes" here just mean adapters that have changes? Because in previous job we did have functionality that would attempt to only run the tests that were necessary to cover code changes.

Reply:
Yes, this is just for adapters. So it is not super helpful for folks working in the core codebase, since it could affect all adapters. That would be pretty neat if we could only run a subset of tests related to actual code changes though.

Reply:
It'll still help for when we make changes to top level md files, or the soon to be merged performance directory.